libcheckout

```diff
@@ -1,5 +1,5 @@
 function checkout() {
-  git clone $SEMAPHORE_GIT_URL $SEMAPHORE_GIT_DIR
+  git clone --branch $SEMAPHORE_GIT_BRANCH --single-branch $SEMAPHORE_GIT_URL $SEMAPHORE_GIT_DIR
```

Comment:
I would like to add some sanity checks to the top of this command like:

```
if [ -z $SEMAPHORE_GIT_BRANCH ]; then
  echo "SEMAPHORE_GIT_BRANCH is not exported"
  exit 1
fi
```

I wrote `exit 1` in the previous command, but I am not 100% sure if that will fly with sourced bash commands.

Reply:
If I am getting it right, you would like to have clear message if we mess up something by not setting all variables required?

Reply:
Yes, but more importantly I would like to avoid weird undesired events.

For example, I am not sure how this would behave:

```
unset SEMAPHORE_GIT_BRANCH
checkout
```

Would it checkout master, or something? It could be dangerous...

Reply:
Ack, thanks for clarifying. I will get back with these checks.

Reply:
Does it make sense to create envcheck tool to check for required variables? It could spit the following output:

```
Exported environment variables:

SEMAPHORE_GIT_BRANCH
SEMAPHORE_JOB_NUMBER
...
```

In case of missing variable it would exit the execution and print the message.

Reply:
I think its better for each script to check its own requirments.

It increases the usability and decreases coupling.

Reply:
To expand on this topic. I would like to keep semaphore as scriptable as possible. For example, someone might hack and do something like this:

```
repos="api cli grpc-gateway"

SEMAPHORE_GIT_BRANCH=master

for repo in $repos; {
   SEMAPHORE_GIT_REPO=github.com/semaphoreci/$REPO.git SEMAPHORE_GIT_DIR=/tmp/repos/$REPO checkout
}
```